code-challenges/objects/challenges-06.test.js

```diff
@@ -0,0 +1,259 @@
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+Use the characters data below for the remainder of the challenges.
+
+Write a function named getHouses that returns a new array containing the names of all of the houses in the data set.
+------------------------------------------------------------------------------------------------ */
+
+let characters = [
+  {
+    name: 'Eddard',
+    spouse: 'Catelyn',
+    children: ['Robb', 'Sansa', 'Arya', 'Bran', 'Rickon'],
+    house: 'Stark'
+  },
+  {
+    name: 'Jon A.',
+    spouse: 'Lysa',
+    children: ['Robin'],
+    house: 'Arryn'
+  },
+  {
+    name: 'Cersei',
+    spouse: 'Robert',
+    children: ['Joffrey', 'Myrcella', 'Tommen'],
+    house: 'Lannister'
+  },
+  {
+    name: 'Daenarys',
+    spouse: 'Khal Drogo',
+    children: ['Drogon', 'Rhaegal', 'Viserion'],
+    house: 'Targaryen'
+  },
+  {
+    name: 'Mace',
+    spouse: 'Alerie',
+    children: ['Margaery', 'Loras'],
+    house: 'Tyrell'
+  },
+  {
+    name: 'Euron',
+    spouse: null,
+    children: [],
+    house: 'Greyjoy'
+  },
+  {
+    name: 'Jon S.',
+    spouse: null,
+    children: [],
+    house: 'Snow'
+  }
+];
+
+const getHouses = (arr) => {
+  let houses = [];
+  // Solution code here...
+  arr.forEach(element => {
+    houses.push(Object.values(element)[3]);
+  });
+  return houses;
+};
+
+/*------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named hasChildrenValues that uses Object.values to determine if any given character in the data set has children.
+
+This function should take in an array of data and a character name and return a Boolean.
+
+For example:
+hasChildrenValues(characters, 'Cersei') will return true
+hasChildrenValues(characters, 'Sansa') will return false
+------------------------------------------------------------------------------------------------ */
+
+const hasChildrenValues = (arr, character) => {
+  // Solution code here...
+  let answer;
+  arr.forEach(element => {
+    if (Object.values(element)[0] === character){
+      if (Object.values(element)[3][0]){
+        answer = true;
+      }
+      else { answer = false }
+    }
+  });
+  return answer;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named hasChildrenEntries that is similar to your hasChildrenValues function from challenge 3, but uses the data's entries instead of its values.
+
+The input and output of this function are the same as the input and output from challenge 3.
+------------------------------------------------------------------------------------------------ */
+
+const hasChildrenEntries = (arr, character) => {
+  // Solution code here...
+  let answer = false;
+  arr.forEach(element => {
+    console.log(Object.entries(element)[0][1] === character && !!Object.entries(element)[2][1][0])

```

Comment:
remember to delete/comment out these console.log()s as it makes the test harder to read.